plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -0,0 +1,230 @@
+    import nevergrad as ng  # type: ignore
+
+    # choice
+
+    if isinstance(description, (ListConfig, list)):
+        choices = [convert_to_deduced_type(x) for x in description]
+        ordered = all(isinstance(c, (int, float)) for c in choices)
+        ordered &= all(
+            c0 <= c1 for c0, c1 in zip(choices[:-1], choices[1:])  # type: ignore
+        )
+        return ng.p.TransitionChoice(choices) if ordered else ng.p.Choice(choices)
+    # custom
+    if isinstance(description, str) and description.startswith(tuple(dir(ng.p))):
+        param: ng.p.Parameter = eval("ng.p." + description)  # pylint: disable=eval-used
+        assert isinstance(param, ng.p.Parameter)
+        return param
+    # scalar
+    if isinstance(description, (dict, DictConfig)):
+        init = ["init", "lower", "upper"]
+        options = init + ["log", "step", "integer"]
+        assert all(x in options for x in description)
+        init_params = {x: y for x, y in description.items() if x in init}
+        if not description.get("log", False):
+            scalar = ng.p.Scalar(**init_params)
+            if "step" in description:
+                scalar.set_mutation(sigma=description["step"])
+        else:
+            if "step" in description:
+                init_params["exponent"] = description["step"]
+            scalar = ng.p.Log(**init_params)
+        if description.get("integer", False):
+            scalar.set_integer_casting()
+        return scalar
+    # constant
+    if isinstance(description, (str, int, float)):
+        return description
+    raise TypeError(f"Unexpected parameter configuration: {description}")
+
+
+class NevergradSweeper(Sweeper):
+    """Returns a Nevergrad parameter from a definition string.
+
+    Parameters
+    ----------
+    config: DictConfig
+      the optimization process configuration
+    version: int
+      version of the API
+    """
+
+    def __init__(
+        self, optim: DictConfig, version: int, parametrization: Optional[DictConfig],
+    ):
+        assert version == 1, "Only version 1 of API is currently available"
+        self.opt_config = optim
+        self.config: Optional[DictConfig] = None
+        self.launcher: Optional[Launcher] = None
+        self.job_results = None
+        self.parametrization: Dict[str, Any] = {}
+        if parametrization is not None:
+            assert isinstance(parametrization, DictConfig)
+            self.parametrization = {
+                x: make_parameter_from_config(y) for x, y in parametrization.items()
+            }
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.launcher = Plugins.instantiate_launcher(
+            config=config, config_loader=config_loader, task_function=task_function
+        )
+
+    def sweep(self, arguments: List[str]) -> None:
+        import nevergrad as ng
```

Comment:
Great. Please add a comment next to imports explaining why this is imported lazily.
This is difficult to test against, I would like to minimize the chance of regression.

Reply:
will do